### PR TITLE
Expand tyAlias when printing nested types

### LINKED ISSUE #8339: Type mismatch error with alias types are useless
```nim
import sequtils

var x: seq[float] = @[1].mapIt(it)
```
Gives the following error message:
`foo.nim(3, 25) Error: type mismatch: got <seq[outType]> but expected 'seq[float]'`

I think `outType` should be resolved and presented to the user, otherwise the error message is completely useless.

I've toyed with this representation:
```
foo.nim(3, 25) Error: type mismatch: got <seq[outType = EXPANDED]> but expected 'seq[float]'
```
and
```
foo.nim(3, 25) Error: type mismatch: got <seq[outType]> but expected 'seq[float]'
Where:
outType = EXPANDED
```
but neither is too appealing to me.

### PR DESCRIPTION
**Pull request by LemonBoy**
Fixes #8339

### CODE REVIEW

**Comment by timotheecour** [comment]
not a blocker for this PR (we can improve later) but:
```nim
type U=type(1.0 * 2.9)
type seq2[V]=seq[V]
type T=seq2[U]
var x: T = @[1].mapIt(it)
```

gives:
```nim
Error: type mismatch: got <seq[outType aka int]> but expected 'T = seq2[U aka float]'
```

any way to improve? eg:
```nim
Error: type mismatch: got <seq[outType aka int]> but expected 'T = seq2[U aka float] aka seq[float]'
```
or, problably simpler:
```nim
Error: type mismatch: got <seq[outType]> aka <seq[int]> but expected 'T = seq2[U] aka seq[float]'
```

**Comment by LemonBoy** [comment]
> E is not expanded with aka

Yep, that's because it's not a `tyAlias`, I have to modify this patch a bit.
That's because `E` is not a 

RE: version1 and version2, I think those have too many `aka`s IMO

**Comment by Araq** [comment]
``aka`` is ugly, better use ``=``, but IMO the compiler should just print the real type, not these silly aliases that only serve to confuse the reader. If the type is more than merely a pointless alias, it would have been a ``distinct`` type in the first place.

**Comment by LemonBoy** [comment]
> the compiler should just print the real type, not these silly aliases that only serve to confuse the reader.

100% agree, but I don't get the logic behind `typeMismatch`. I'd just always print the expanded type and call it a day.

**Comment by Araq** [comment]
There is some logic here in order to avoid excessively long error messages, but I don't think it affects the piece of code you're patching.

**Comment by timotheecour** [comment]
> RE: version1 and version2, I think those have too many akas IMO

version2 has exactly as many `aka` (or `=` if you prefer) as your PR; it's basically whole term rewriting instead of partial rewriting, when there is some form of aliasing involved.

> the compiler should just print the real type, not these silly aliases that only serve to confuse the reader.

possible. Curious whether there'd be any case where it'd lead to less helpful messages; we could control this via a `hint[mismatchVerboseType]` (with a better hint name).

with default:
```nim
Error: type mismatch: got <proc (x: float){.noSideEffect, gcsafe, locks: 0.}> but expected <proc (x: char){.closure.}>
```
with `--hints[mismatchVerboseType]`:
```nim
Error: type mismatch: got <proc (x: MyFloat){.noSideEffect, gcsafe, locks: 0.}> aka <proc (x: float){.noSideEffect, gcsafe, locks: 0.}> but expected <proc (x: E){.closure.}> aka <proc (x: char){.closure.}>
```

File: compiler/types.nim [hunk 1]
```diff
@@ -426,6 +426,8 @@ proc typeToString(typ: PType, prefer: TPreferedDesc = preferName): string =
        sfAnon notin t.sym.flags:
     if t.kind == tyInt and isIntLit(t):
       result = t.sym.name.s & " literal(" & $t.n.intVal & ")"
+    elif t.kind == tyAlias:
+      result = t.sym.name.s & " aka " & typeToString(t.sons[0])
```
Comment (Araq) [inline compiler/types.nim]: Please change this to

```nim
  result = typeToString(t.sons[0])
```

**Comment by LemonBoy** [comment]
Amended and ready to go, we'll fix the `proc` case another day (or in another life)

**Pull request merged**